### ISSUE TITLE
[Enhancement] Interaction State Utility - Hover

### ISSUE THREAD

**Issue by jkaltoft** [kirbydesign/designsystem#1935]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement

I would like a styling utility that can be applied to all elements/components that can be hovered. Ideally it should be prepared for handling _all_ interaction states (hover, press, focus), but this issue should just implement hover.

This issue derives from the research made in #1866

## Describe the solution you'd like

Inspired by [Material Design's interaction states](https://m3.material.io/foundations/interaction-states), this utility should create a state layer that can have varying degrees of opacity. The color should be derived from its content or configurable, cf. Material Design's "On colors".

However, the actual implementation should not rely on an empty DOM element if possible. We prefer to implement the state layer as either a simple change of background color or as a pseudo element. More research is needed. 

Experiment with HSLA color, brightness() CSS filter, opacity, pseudo element solution etc.

- [x] Create new PoC
- [x] Should we offer option to configure opacity degree and "on color"?

This issue does not concern elements/components with non-solid backgrounds, e.g. image as background. Those cases should be handled individually when/if we bump into any issues in the future.

We should try to implement the solution using utility mixins as suggested in [comments on the research issue](https://github.com/kirbydesign/designsystem/issues/1866#issuecomment-984482799).

The solution could end up with 2 ways to implement state layers. We may need both implementations:

### Simple background color change

For all elements/components including self-closing/singleton/atomic ones.

- Can _not_ contain other components, e.g. `<input />` (text, checkboxes, radio buttons etc.)
- Implement solution as a calculated background color based on opacity and "on color"
- Utility can only be applied to elements/components with fully opaque backgrounds (solid color)

### Empty DOM element (or pseudo element)

📣 UPDATE: Implementation should be moved to a separate issue

For container elements/components.

- Can contain other components, e.g. `<div>` and `<kirby-card>`
- Implement an actual layer (DOM element) - see [PoC](https://codepen.io/jkaltoft/pen/gOGrEON?editors=0100)
- Utility can be applied to elements/components with _all_ kinds of backgrounds, including images, gradients and completely transparent backgrounds

## Subtasks

Add subtasks as they emerge during research.

Some "spin-off tasks" may be out of scope. Take them into consideration and either solve them, plan them for later or discard them (before closing this issue).

- [x] #1992
- [x] Create `apply-hover()` mixin
    - [x] Verify parameter validity
    - [x] Determine how to handle transparent background
- [x] Utilize `apply-hover()`, e.g., in `<Button>` component
    - Needs to be applied multiple places - every variant, ie., attention levels, danger, disabled etc.
- [ ] Organize/re-structure non-component stylesheets, e.g., using ITCSS or similar methodology
- [x] ~~Create color converter - e.g. as Sass function or mixin - that can at least convert HEX and RGB colors to HSL. Use the Sass color module's functions.~~
    - ~~Alternative A: Add HSL variants of color custom properties~~
    - ~~Alternative B: Add function that can create HSL variants where needed~~
- ~~Refactor CSS custom properties for colors and related Sass variables/functions/mixins~~
- [x] Create design tokens for "loudness"
- [ ] #2006
- [ ] #2008
- [ ] #2014
- [ ] #2015
- [x] #2016

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [x] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [x] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [x] ~~Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".~~
- [x] ~~Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.~~

### Review
Once the issue has been implemented and is ready for review:

- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [x] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by jkaltoft** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-Owner-Workspace-602b70d809e4aa00141a1a93/issues/kirbydesign/designsystem/1935?planning-poker) with ZenHub @MadsBuchmann @RasmusKjeldgaard @mictro

**Comment by jkaltoft** [comment]
- [ ] Create color converter - probably as Sass function or mixin - that can at least convert HEX and RGB colors to HSL

**Comment by jkaltoft** [comment]
The "pseudo element" solution for elements/components with non-solid backgrounds should be done in a separate issue - ideally when implementing hover for a component that actually needs it. Perhaps `<Card>` is a good candidate?
